src/main/java/com/easypost/model/beta/Referral.java

```diff
@@ -0,0 +1,303 @@
+        Map<String, String> params = new HashMap<>();
+        params.put("number", number);
+        params.put("exp_month", String.valueOf(expirationMonth));
+        params.put("exp_year", String.valueOf(expirationYear));
+        params.put("cvc", cvc);
+
+        URL stripeUrl = new URL("https://api.stripe.com/v1/tokens");
+        HttpURLConnection conn = (HttpURLConnection) stripeUrl.openConnection();
+        conn.setRequestMethod("POST");
+        conn.setRequestProperty("Authorization", String.format("%s %s", "Bearer", easypostStripeApiKey));
+        conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+        conn.setDoOutput(true);
+
+        String encodedURL = getEncodedURL(params, "card");
+        byte[] postData = encodedURL.getBytes(StandardCharsets.UTF_8);
+
+        try (DataOutputStream wr = new DataOutputStream(conn.getOutputStream())) {
+            wr.write(postData);
+        }
+
+        StringBuilder response;
+
+        try (BufferedReader br = new BufferedReader(
+                new InputStreamReader(conn.getInputStream()))) {
+
+            String line;
+            response = new StringBuilder();
+
+            while ((line = br.readLine()) != null) {
+                response.append(line);
+                response.append(System.lineSeparator());
+            }
+            br.close();
+        } finally {
+            conn.disconnect();
+        }
+
+        String responseBody = response.toString();
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> responseMap = GSON.fromJson(responseBody, Map.class);
+
+        return responseMap.get("id").toString();
+    }
+
+    /**
+     * Submit Stripe credit card token to EasyPost.
+     *
+     * @param referralApiKey API key of the referral user.
+     * @param stripeObjectId Stripe token.
+     * @param priority Credit card priority.
+     *
+     * @return CreditCard object.
+     * @throws EasyPostException when the request fails.
+     */
+    private static CreditCard createEasypostCreditCard(String referralApiKey,
+        String stripeObjectId, String priority) throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+        params.put("stripe_object_id", stripeObjectId);
+        params.put("priority", priority);
+
+        Map<String, Object> wrappedParams = new HashMap<>();
+        wrappedParams.put("credit_card", params);
+
+        return request(RequestMethod.POST, String.format("%s/%s", EasyPost.BETA_API_BASE, "credit_cards"),
+        wrappedParams, CreditCard.class, referralApiKey);
+    }
+
+    /**
+     * Create Encoded URL from a Map.
+     *
+     * @param params Map of parameters to be encoded.
+     * @param parentKey Parent key in the encoded URL.
+     * @return Encoded URL for Stripe API call.
+     * @throws Exception
+     */
+    private static String getEncodedURL(Map<String, String> params, String parentKey)
```

Comment:
Let's move this to a `Util` file because it's by nature a utility for encoding URLs and can be reused in the future if necessary. Since we are adding it brand new, let's put it in the right place to begin with.